File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/params/set_cartesian_control_params.py
#!/usr/bin/env python3

import humanoid_sdk_py
import humanoid_sdk_py.h10w as h10w
import time


def main():
    print("=== 设置笛卡尔空间控制参数示例 ===")
    params = h10w.H10wParams()
    time.sleep(1)

    kp6 = [10000.0] * 6
    kd6 = [300.0] * 6
    kpv6 = [800.0] * 6
    kp6_r = [10500.0] * 6
    kd6_r = [320.0] * 6
    kpv6_r = [850.0] * 6

    set_params = [
        (int(h10w.CartIndex.LEFT_ARM), kp6, kd6, kpv6),
        (int(h10w.CartIndex.RIGHT_ARM), kp6_r, kd6_r, kpv6_r),
    ]

    ret = params.setCartesianControlParams(set_params)
    if ret != 0:
        print("设置笛卡尔空间控制参数失败")
        return -1
    print("设置笛卡尔空间控制参数成功")
    return 0


if __name__ == "__main__":
    exit(main())

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/params/set_cartesian_control_params.cpp
// set_cartesian_control_params.cpp

#include <array>
#include <iostream>

#include "common/common_def.h"
#include "common/h10w_def.h"
#include "h10-w/h10w_params.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 设置笛卡尔空间控制参数示例 ===" << std::endl;
  H10wParams params;

  CartesianControlParams set_params = {
      {static_cast<uint32_t>(CartIndex::LEFT_ARM),
       std::array<double, 6>{10000.0, 10000.0, 10000.0, 10000.0, 10000.0,
                             10000.0},
       std::array<double, 6>{300.0, 300.0, 300.0, 300.0, 300.0, 300.0},
       std::array<double, 6>{800.0, 800.0, 800.0, 800.0, 800.0, 800.0}},
      {static_cast<uint32_t>(CartIndex::RIGHT_ARM),
       std::array<double, 6>{10500.0, 10500.0, 10500.0, 10500.0, 10500.0,
                             10500.0},
       std::array<double, 6>{320.0, 320.0, 320.0, 320.0, 320.0, 320.0},
       std::array<double, 6>{850.0, 850.0, 850.0, 850.0, 850.0, 850.0}},
  };

  if (params.setCartesianControlParams(set_params) != 0) {
    std::cerr << "设置笛卡尔空间控制参数失败" << std::endl;
    return -1;
  }

  std::cout << "设置笛卡尔空间控制参数成功" << std::endl;
  return 0;
}
